build.sh

```diff
@@ -1 +1,6 @@
-make clean ; ./autogen.sh && ./nomacro.pl && ./configure CFLAGS="-O3" && make
+# Intel or AMD Ryzen or ARM-aarch64
+make clean ; \
+./autogen.sh && \
+./nomacro.pl && \
+./configure CFLAGS="-O3" && \
```

Comment:
I recommend `-O2 -fomit-frame-pointer` instead of `-O3`, although the difference is minor (and expected to be seen mostly on 32-bit x86 builds).

Reply:
thanks so much! 😆

Reply:
@solardiz 

```bash
./configure CFLAGS="-O2 -fomit-frame-pointer -msse2"
```
it works well on intel, but build error on `aarch64`. :sob:   "-O3" was no issue :thinking: 

```bash
make[1]: Leaving directory '/home/pi/cpuminer-mc-yespower'
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking target system type... aarch64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/home/pi/cpuminer-mc-yespower':
configure: error: C compiler cannot create executables
See `config.log' for more details
pi@kirari:~/cpuminer-mc-yespower$
```

any idea?

Reply:
info about gcc & cpu

```bash
$ gcc --version
gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```bash
$ cat /proc/cpuinfo
processor    : 0
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 1
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 2
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 3
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 4
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 5
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 6
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3

processor    : 7
BogoMIPS    : 19.84
Features    : fp asimd aes pmull sha1 sha2 crc32
CPU implementer    : 0x41
CPU architecture: 8
CPU variant    : 0x0
CPU part    : 0xd03
CPU revision    : 3
```

```bash
$ lscpu
Architecture:          aarch64
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             2
CPU max MHz:           1400.0000
CPU min MHz:           400.0000
```

Reply:
@solardiz it solved `without -msse2` and a few improved on `aarch64` :smile: 

```
make clean && ./autogen.sh && ./configure CFLAGS="-O2 -fomit-frame-pointer" && make
```

0.560 kH/s >> 0.569 kH/s 
`1.5%` additional boost! great thanks!